#8: Adiciona funcionalidades ao menu da biblioteca

lfsevergnini opened the pull request.

ramirosevergnini committed:
Switch case em andamento!

lfsevergnini commented on src/bibliotecaprojeto/Biblioteca.java:
```diff
@@ -57,12 +56,40 @@ public Emprestimo realizarEmprestimo(Biblioteca b1, Usuario usuario, Livro livro
             livros.remove(livro);
+            return livro;
         }
+        return null;
```
Parece que não tem como esse método funcionar 🤔.

Para utilizar o `contains()`, em vez de receber o o título,  você precisa receber de fato um `Livro livro`.

Porém, pode ser que o Java tenha evoluído de alguma forma e isso funcione, então me avisa aí se eu estiver enganado.

lfsevergnini commented on src/bibliotecaprojeto/Biblioteca.java:
```diff
@@ -57,12 +56,40 @@ public Emprestimo realizarEmprestimo(Biblioteca b1, Usuario usuario, Livro livro
+        }
+        return null;
+    }
+    public Usuario confirmarCadastro(String tecladoUsuario){
```
Olhando somente para o nome dos métodos, não consigo entender qual é o propósito deles.

Confirmar? Parece que é uma busca, na verdade.

lfsevergnini commented on src/bibliotecaprojeto/Biblioteca.java:
```diff
@@ -57,12 +56,40 @@ public Emprestimo realizarEmprestimo(Biblioteca b1, Usuario usuario, Livro livro
     }
-    public void sair(){
+    public String colocarSenha(String admSenha){
+        String senha = "723109";
```
👀

lfsevergnini commented on src/bibliotecaprojeto/BibliotecaProjeto.java:
```diff
@@ -15,21 +15,24 @@ public static void main(String[] args) {
+        System.out.println("Você deseja entrar em nossa biblioteca? (sim) ou (nao)");
         c = whileCond.nextLine();
-        while (c != "sim"){
+        System.out.println("Seja muito bem vindo a biblioteca Severgnini Cultura!");
```
📚 🧐

lfsevergnini commented on src/bibliotecaprojeto/BibliotecaProjeto.java:
```diff
@@ -76,44 +133,25 @@ public static void main(String[] args) {
             
 //        Remove livros do sistema
-//        b1.removerLivro(l1);
+        b1.removerLivro(l1);
```
Não entendi pq tá chamando isso, pode ter sobrado de algum teste antigo.

lfsevergnini commented on src/bibliotecaprojeto/BibliotecaProjeto.java:
```diff
@@ -15,21 +15,24 @@ public static void main(String[] args) {
-        while (c != "sim"){
+        System.out.println("Seja muito bem vindo a biblioteca Severgnini Cultura!");
+        loopPrincipal:
+        while (!"nao".equals(c)){
```
Você nunca mais lê o valor de `c`, então isso é _praticamente_ um `while(true)`.

Edit: vi que você tá usando um break com loop nomeado lá embaixo. VOu deixar uma sugestão lá.

lfsevergnini commented on src/bibliotecaprojeto/BibliotecaProjeto.java:
```diff
@@ -76,44 +133,25 @@ public static void main(String[] args) {
-                default:
+                case "7":
+                    System.out.println("Saindo da biblioteca...");
+                    break loopPrincipal;
```
Apesar de isso funcionar na prática, não me parece uma boa ideia, no geral.

Em vez disso, considere a seguinte possiblidade:
* Esqueça a questão de "deseja entrar na biblioteca"... Pode remover essa parte.
* Crie uma variável `boolean sair = false` e teste ela na condição do seu `while(!sair)`.
* Caso o usuário digite a opção 7 (ou qualquer que seja a opção de sair), você assinala `sair = true`.

Parece melhor?

ramirosevergnini replied:
Opa, vou fazer as alterações necessárias, qualquer dúvida te chamo!!

lfsevergnini reviewed (commented):
@ramirosevergnini deixei uns comentários aí.

Está indo muito bem!

Peço que considere algumas das sugestões. Caso não faça sentido, avisa aí.

Valeu!

ramirosevergnini committed:
Adição de cases e alteração da forma de leitura dos metodos

Ainda não está pronto, então está dando erro, mas segunda resolvo o que falta e não é para dar erro hehe

ramirosevergnini committed:
Métodos login, registrar em andamento...

ramirosevergnini committed:
Junção dos métodos de cadastro.

ramirosevergnini committed:
WIP: Reformulação do loop do programa
